addons/l10n_de/models/ir_attachment.py

```diff
@@ -24,11 +24,20 @@ def _except_audit_trail(self):
         for attachment in audit_trail_attachments:
             move = id2move.get(attachment.res_id)
             if move and move.posted_before and move.country_code == 'DE':
-                raise UserError(_("You cannot remove parts of the audit trail."))
+                ue = UserError(_("You cannot remove parts of the audit trail."))
+                ue.__audit_trail = True
+                raise ue
 
     def write(self, vals):
         if vals.keys() & {'res_id', 'res_model', 'raw', 'datas', 'store_fname', 'db_datas'}:
-            self._except_audit_trail()
+            try:
+                self._except_audit_trail()
+            except UserError as e:
+                if hasattr(e, '_IrAttachment__audit_trail') and vals.get('res_model') == 'documents.document':
```

Comment:
:grimacing:

Reply:
Yeah, I could not have it a private variable, but it would be less safe for server actions.
It's already bypassable easily by changing the company currency anyway so I might just do it.